yarn-project/acir-simulator/src/public/executor.ts

```diff
@@ -119,6 +120,15 @@ export class PublicExecutor {
         this.log(`Returning from nested call: ret=${childExecutionResult.returnValues.join(', ')}`);
         return padArrayEnd(childExecutionResult.returnValues, Fr.ZERO, NOIR_MAX_RETURN_VALUES).map(toACVMField);
       },
+      emitUnencryptedLog: ([...chars]: ACVMField[]) => {
+        let unencryptedLog = '';
+        for (const hex of chars) {
+          const unicodeChar = String.fromCharCode(parseInt(hex, 16));
```

Comment:
I feel like using the string type here is a bit weird given that the native type of logs should be bytes. I wrote the `convertACVMFieldToBuffer` function which directly converts the field to a buffer. I think it would make sense to use it here. Can you give it a try?

Reply:
ah yes I didn't push the change after our discussion that this should take generic bytes. I had this because it's more efficient with strings (Gets rid of all the 0s that are included in each character).